WooCommerce/Classes/ViewRelated/Orders/Order Creation/Synchronizer/OrderSynchronizer.swift

```diff
@@ -69,6 +69,10 @@ protocol OrderSynchronizer {
     ///
     var setProduct: PassthroughSubject<OrderSyncProductInput, Never> { get }
 
+    /// Sets multiple products with their quantities.
+    ///
+    var setProducts: PassthroughSubject<[OrderSyncProductInput], Never> { get }
```

Comment:
Now that we can add an array of products, can we remove `setProduct` and replace its usage with `setProducts` passing an array of one item? That way we remove non necessary code

Reply:
I'd prefer to leave both implementations ( setProduct, and setProducts ) at the moment for several reasons:

- Keep this PR smaller.
- We can "encapsulate" and use setProducts specifically only on multi-selection logic, which is under feature flag and beta flag, leaving single-selection as it is.
- There's 30+ tests that need to be adapted.
- The logic revolving updating OrderItems that behaves slightly different in each case, so would require some refactor as well on different helper functions.

I've opened a new issue here https://github.com/woocommerce/woocommerce-ios/issues/9195 to replace its usage in a future PR to simplify code 👍

Reply:
I think that if we are merging to `trunk` we should try to leave the code as clean as possible for now, avoiding duplicated code. If the PR gets bigger we can split it into smaller ones, but I don't think creating technical debt is a good idea if we can address the problem right away.

Reply:
@iamgabrielma thanks for the call in Slack. After we checked the logic is different for one and multiple items, It's ok to leave these, but we should at least remove those pieces of duplicated code other than that logic.

Reply:
>  It's ok to leave these, but we should at least remove those pieces of duplicated code other than that logic.

Great! I extracted the usage for `updateOrderItems()` into a different function in https://github.com/woocommerce/woocommerce-ios/pull/9173/commits/51df0918d29fea785e9b76d7627a141960262d9e , so we could remove the same code from 3 different places. 

I had to use an `inout` parameter for one of the inputs, as we run this inside of a loop and I couldn't pass it as a constant, let me know if there's any concern with this approach. I confirmed that keeps working as expected and all test pass correctly.